### PR TITLE
Vr merges

### PR DESCRIPTION
**Pull request by putskan**

### CODE REVIEW

File: server/server/posts/api/serializers.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            "author_profile_picture",
+        ]
+
+    def get_author_profile_picture(self, obj):
+        return InstructorProfile.objects.get(user=obj.author).profile_picture
+
+    def get_images(self, obj):
+        request = self.context.get("request")
+        return [
+            request.build_absolute_uri(img.image_url.url)
```
[the remark this answers is not in the document]
Comment (putskan) [reply server/server/posts/api/serializers.py:62]: get image urls from serializer instead of building

File: server/server/posts/apps.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+
+class PostsConfig(AppConfig):
+    name = "server.posts"
+    verbose_name = _("Posts")
+
+    def ready(self):
+        try:
+            import server.posts.signals  # noqa F401
+        except ImportError:
```
Comment (putskan) [inline server/server/posts/apps.py:12]: supress

File: server/server/posts/models.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+from server.utils.model_fields import random_slug
+
+
+class Post(models.Model):
+    slug = models.CharField(
+        max_length=40,
+        default=random_slug,
+        unique=True,
+    )
+    creation_time = models.DateTimeField(
```
Comment (putskan) [inline server/server/posts/models.py:14]: created instead

File: server/server/posts/tests.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
```
Comment (putskan) [inline server/server/posts/tests.py:1]: add tests

File: server/server/schools/models.py [hunk 5]
```diff
@@ -16,7 +16,7 @@ class School(models.Model):
     name = models.CharField(max_length=50)
     address = models.CharField(max_length=50)
     address_city = models.CharField(max_length=50)
-    zip_city = models.CharField(max_length=15)
+    zip_city = models.CharField(max_length=15, blank=True)
```
Comment (putskan) [inline server/server/schools/models.py:19]: change to address_zip

File: server/server/users/api/views.py [hunk 6]
```diff
@@ -121,12 +126,53 @@ def get_queryset(self):
 
     @action(detail=False, methods=["POST"])
     def bulk_create(self, request):
```
Comment (putskan) [inline server/server/users/api/views.py:128]: test CSV upload

Comment (putskan) [reply server/server/users/api/views.py:128]: bytesIo, stringIo

File: server/server/users/api/views.py [hunk 7]
```diff
@@ -144,15 +190,78 @@ def get_queryset(self):
...
-        serializer = ManageCoordinatorsSerializer(
-            data=request.data, context={"request": request}, many=True
-        )
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        users_to_invite = []
+        if request.FILES:
```
Comment (putskan) [inline server/server/users/api/views.py:194]: flatten if statement (check negation)

File: server/server/users/api/views.py [hunk 8]
```diff
@@ -144,15 +190,78 @@ def get_queryset(self):
...
+                            "email": email,
+                        }
+                    )
+                    emails.add(email)
+
+            # remove already existing emails
+            already_existing_emails = User.objects.filter(
+                email__in=[user["email"] for user in users_to_invite]
+            ).values_list("email", flat=True)
+            users_to_invite = [
```
Comment (putskan) [inline server/server/users/api/views.py:222]: generator expression instead?

File: server/server/users/api/views.py [hunk 9]
```diff
@@ -144,15 +190,78 @@ def get_queryset(self):
...
+                return Response(serializer.data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
+class ExportConsumerListViewSet(ModelViewSet):
+    permission_classes = [AllowCoordinator]
+    serializer_class = ManageConsumersSerializer
+    lookup_field = "slug"
+    renderer_classes = (UsersCSVRenderer,)
+    results_field = "results"
```
Comment (putskan) [inline server/server/users/api/views.py:244]: remove?

**Review by thehiveril — APPROVED** [review]

File: server/server/users/api/views.py (hunk 9 above)
Comment (putskan) [reply server/server/users/api/views.py]: yes

File: server/server/users/api/views.py (hunk 8 above)
Comment (putskan) [reply server/server/users/api/views.py]: drf doesnt like generators

**Pull request merged**